### PR TITLE
Merge feat/CCN3-623/application-submission into develop

### PR DESCRIPTION
**Pull request by ricaduchateau**
## Description

Building the 'Next steps' pages, accessible from final confirmation/application submission page, as well as accompanying unit tests. Resolves [CCN3-623].

## Todo's before PR

- [x] Rebase from develop
- [x] Unit tests passed (`make test`)
- [x] PR naming according our [guide](https://github.com/InformedSolutions/OFS-MORE-DevOps-Tooling/wiki/Commits-guideline)
- [x] PR description describes the overall goals of PR commits
- [x] Templates been checked with relevant user-story

## PR Todo's

Please refer to PR [guide](https://github.com/InformedSolutions/OFS-MORE-DevOps-Tooling/wiki/Pull-requests#how-to-submit-a-pull-request)

- [x] Specified reviewers (even if it's yourself)
- [x] Specified assigneses (those who'll merge)
- [x] Specified labels
- [x] Specified milestone

## PR review

Please refer to PR review [guide](https://github.com/InformedSolutions/OFS-MORE-DevOps-Tooling/wiki/Pull-requests#how-to-review-a-pull_request)

- [x] Code syntax formatting checked
- [x] Debug messages are absent

## PR merge

Please refer to PR merge [guide](https://github.com/InformedSolutions/OFS-MORE-DevOps-Tooling/wiki/Pull-requests#how-to-merge-a-pull_request)

### CODE REVIEW

File: application/test_views.py [hunk 1]
```diff
@@ -1310,6 +1313,42 @@ def test_page_not_displayed_without_id(self):
...
+            self.assertEqual(0, 0)
+
+    def test_url_resolves_to_page(self):
+        found = resolve(settings.URL_PREFIX + '/next-steps/interview/')
+        self.assertEqual(found.func, prepare_for_interview)
+
+    def test_page_not_displayed_without_id(self):
+        c = Client()
+        try:
+            c.get(settings.URL_PREFIX + '/next-steps/interview?id=')
```
Comment (holms) [inline application/test_views.py]: url_reverse()?

File: application/views/base.py [hunk 2]
```diff
@@ -3632,6 +3633,95 @@ def payment_confirmation(request):
...
+    if request.method == 'GET':
+        application_id_local = request.GET['id']
+        order_code = Application.objects.get(pk=application_id_local).order_code
+        form = DocumentsNeededForm()
+        variables = {
+            'application_id': application_id_local,
+            'order_code': order_code,
+            'form': form
+        }
+        return render(request, 'next-steps-documents.html', variables)
```
Comment (holms) [inline application/views/base.py]: Line break :)

File: application/views/base.py [hunk 3]
```diff
@@ -3632,6 +3633,95 @@ def payment_confirmation(request):
...
+    if request.method == 'GET':
+        application_id_local = request.GET['id']
+        order_code = Application.objects.get(pk=application_id_local).order_code
+        form = PrepareForInterviewForm()
+        variables = {
+            'application_id': application_id_local,
+            'order_code': order_code,
+            'form': form
+        }
+        return render(request, 'next-steps-interview.html', variables)
```
Comment (holms) [inline application/views/base.py]: Line break

File: application/views/base.py (hunk 2 above)
Comment (ricaduchateau) [reply application/views/base.py]: Hi @holms, this should now be fixed in the next commit. With kind regards, Rica

File: application/views/base.py (hunk 3 above)
Comment (ricaduchateau) [reply application/views/base.py]: Hi @holms, this should now be fixed in the next commit. With kind regards, Rica

File: application/test_views.py (hunk 1 above)
Comment (ricaduchateau) [reply application/test_views.py]: Hi @holms, this should now be fixed in the next commit. With kind regards, Rica

**Comment by holms** [comment]
@ricaduchateau is TODO list is passed?

**Comment by ricaduchateau** [comment]
Hi @holms, I've checked all the items in the to-do list. With kind regards, Rica

**Comment by cruddasj** [comment]
Are there any outstanding actions on this PR or can this be merged?

**Comment by ricaduchateau** [comment]
Hi @cruddasj,

I think @GeevesH-IS is still reviewing the built pages against the mock-ups.

With kind regards,

Rica

**Comment by ricaduchateau** [comment]
Hi @holms,

The merge conflicts on this pull request have now been fixed.

With kind regards,

Rica

File: application/templates/next-steps-documents.html [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+            </table>
+            <br>
+            <br>
+            <form method="post" novalidate {% if form.is_multipart %}enctype="multipart/form-data" {% endif %}>
+                {{form.as_div}}
+                {% csrf_token %}
+                <input type="submit" class="button" value="Continue">
+                <input type="hidden" id="id" name="id" value="{{application_id}}"/>
+            </form>
+        </div>
```
Comment (GeevesH-IS) [inline application/templates/next-steps-documents.html]: On the mockup, there's another back button below the continue/submit button

File: application/templates/next-steps-home.html [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+{% block page_title %}Get your home ready{% endblock %}
+{% load static %}
+{% load govuk_template_base %}
+
+{% block inner_content %}
+
+<div class="grid-row">
+    <div class="column-two-thirds">
+        <!-- Back button -->
+        <a href="{{ URL_PREFIX }}/next-steps/documents?id={{application_id}}" class="link-back"
```
Comment (GeevesH-IS) [inline application/templates/next-steps-home.html]: On the mockup, there actually isn't a back button on this page, but I can't see why considering there's one on every single other page. Thought I'd double check

Comment (GeevesH-IS) [reply application/templates/next-steps-home.html]: Discussed and approved with @ricaduchateau

File: application/templates/next-steps-documents.html (hunk 4 above)
Comment (ricaduchateau) [reply application/templates/next-steps-documents.html]: Hi @GeevesH-IS,

As discussed, the mock-up just has bad formatting.

With kind regards,

Rica

File: application/templates/next-steps-home.html (hunk 5 above)
Comment (ricaduchateau) [reply application/templates/next-steps-home.html]: Hi @GeevesH-IS,

As discussed, the mock-up just has bad formatting.

With kind regards,

Rica

**Pull request merged**